src/mlpack/methods/ann/layer/multihead_attention_impl.hpp

```diff
@@ -145,30 +141,45 @@ Forward(const InputType& input, OutputType& output)
 
   // Apply the attention mask if provided. The attention mask is used to black-
   // out future sequences and generally used in Encoder-Decoder attention.
-  // The attention mask has elements 0 or -infinity.
+  // The attention mask has elements 0 or 1.
   // The shape of the attention mask : (tgtSeqLen, srcSeqLen).
   if (!attnMask.is_empty())
   {
     if (attnMask.n_rows != tgtSeqLen || attnMask.n_cols != srcSeqLen)
       Log::Fatal << "The size of the 'attn_mask' is not correct.\n";
-    scores.each_slice() += attnMask;
+    // not sure if there is a better way to do this.  now that the mask
+    // is 0 or 1, we can't simply add, but converting first seems slow
+    const arma::uword rows = tgtSeqLen;
+    const arma::uword cols = srcSeqLen;
+    for (arma::uword i=0; i<rows; i++) {
+      for (arma::uword j=0; j<cols; j++) {
```

Comment:
@akropp Suggest to swap these two loop declarations (the cols loop should be the outer loop).  Armadillo stores data in column major format, so it's more efficient to traverse down each column rather than jump across columns.

Reply:
We decided to switch it back to using (-Inf,0) for the masks for computational efficiency (and to match the literature), so don't have to do this anymore.  Thanks for the suggestion though.